cfme/infrastructure/deployment_roles.py

```diff
@@ -41,6 +42,24 @@ def __init__(self, name, provider, appliance=None):
         self.name = name
         self.provider = provider
 
+    def get_detail(self, *ident):
+        """ Gets details from the details InfoBlock
+        Args:
+            *ident: An InfoBlock title, followed by the Key name,
+                e.g. "Relationships", "All VMs"
+        Returns: A string representing the contents of the InfoBlock's value.
+        """
+        navigate_to(self, 'Details')
+        return details_page.infoblock.text(*ident)
+
+    def delete(self):
+        navigate_to(self, 'Details')
+        menu_item = 'Remove item'
```

Comment:
**Optional**
Consider using a dictionary here and calling pick, its a good bit cleaner.